### PR TITLE
Move `kolena.workflow.metrics` to `kolena.metrics`

### PR DESCRIPTION
**Pull request by consoull**
### Linked issue(s):
Fixes KOL-4727

### What change does this PR introduce and why?
Moves `kolena.workflow.metrics` to `kolena.metrics`. The metrics package is not specific to workflows so it is being relocated. `kolena.workflow.metrics` remains for backwards compatibility. 

### Please check if the PR fulfills these requirements

- [x] Include reference to internal ticket and/or GitHub issue "Fixes #NNNN" (if applicable)
- [x] Relevant tests for the changes have been added
- [x] Relevant docs have been added / updated

### CODE REVIEW

File: docs/reference/workflow/metrics.md [hunk 1]
```diff
@@ -1,3 +0,0 @@
-# `kolena.workflow.metrics`
```
Comment (yifanwu-kolena) [inline docs/reference/workflow/metrics.md:1]: i think we should still keep this doc for backward compatibility

File: mkdocs.yml [hunk 2]
```diff
@@ -64,7 +65,6 @@ nav:
         - Assets: reference/workflow/asset.md
         - Plots: reference/workflow/plot.md
         - <code>define_workflow</code>: reference/workflow/define-workflow.md
-        - <code>metrics</code>: reference/workflow/metrics.md
```
Comment (yifanwu-kolena) [inline mkdocs.yml]: ditto.

**Review by yifanwu-kolena — APPROVED** [review]
one comment for doc backward compatibility, otherwise LGTM

File: docs/reference/workflow/metrics.md (hunk 1 above)
Comment (consoull) [reply docs/reference/workflow/metrics.md:1]: This results in breaking relative links in metrics docstrings. Will require additional changes to support

**Review by yifanwu-kolena — APPROVED** [review]

**Pull request merged**